documentation/examples/custom-sd/adapter/adapter.go

```diff
@@ -72,12 +99,22 @@ func (a *Adapter) generateTargetGroups(allTargetGroups map[string][]*targetgroup
 			for name, value := range group.Labels {
 				newLabels[string(name)] = string(value)
 			}
-			// Make a unique key, including the current index, in case the sd_type (map key) and group.Source is not unique.
-			key := fmt.Sprintf("%s:%s:%d", k, group.Source, i)
-			tempGroups[key] = &customSD{
+
+			sdGroup := customSD{
 				Targets: newTargets,
 				Labels:  newLabels,
 			}
+			// Make a unique key, including the current index, in case the sd_type (map key) and group.Source is not unique.
```

Comment:
this comment should be updated based on your changes

Reply:
done